### PR TITLE
Feat/helpers access

### LINKED ISSUE #123: Register helpers to vue instance
This way, we can call `this.$logout` from within any component and without imports

### PR DESCRIPTION
**Pull request by jb3rndt**
### Changes made by this Pull Request

- Converts store to a plugin and registers it on the app

### CODE REVIEW

File: static/src/main.ts [hunk 1]
```diff
@@ -24,4 +31,38 @@ app.use(Toaster);
...
+    component: LoginView,
+    beforeEnter: () => {
+      return store.isLoggedIn ? "/tasks" : true;
+    },
+  },
+  {
+    path: "/tasks",
+    component: TasksOverView,
+    beforeEnter: () => {
+      return store.isLoggedIn ? true : "/login";
```
Comment (jb3rndt) [inline static/src/main.ts]: We could replace `store` with `(app.config.globalProperties.$store as StoreApi)`

File: static/src/App.vue [hunk 2]
```diff
@@ -12,27 +11,26 @@ export default defineComponent({
...
       task: undefined as Task | undefined,
       ready: false,
-      store,
     };
   },
 
   async mounted() {
-    store.initialize(this.$cookies);
-    if (store.jwt) {
+    this.$store.initialize(this.$cookies);
```
Comment (bewee) [inline static/src/App.vue]: Can we simply remove this line?

File: static/src/main.ts [hunk 3]
```diff
@@ -15,7 +15,8 @@ import LoginView from "./components/LoginView.vue";
 import TasksOverView from "./components/TasksOverView.vue";
 import TaskView from "./components/TaskView.vue";
 import RedirectView from "./components/RedirectView.vue";
-import { store } from "@/utils/store";
+import Store from "@/utils/store";
+import {store} from "@/utils/store";
```
Comment (bewee) [inline static/src/main.ts]: ?

File: static/src/utils/store.ts [hunk 4]
```diff
@@ -1,20 +1,21 @@
...
 
   initialize(cookies: VueCookies) {
     this.$cookies = cookies;
     this.jwt = cookies.get("jwt");
   },
 
-  get jwt() {
+  get jwt(): string | undefined {
     return this._jwt;
   },
```
Comment (bewee) [inline static/src/utils/store.ts]: Check again whether all this additional typing information is really necessary

**Review by bewee — APPROVED** [review]
Nice nice!

File: static/src/main.ts (hunk 1 above)
Comment (jb3rndt) [reply static/src/main.ts]: We'll keep it as is

**Pull request merged**